### PR TITLE
s3: support anonymous access

### PR DESCRIPTION
**Pull request by 201341**
Signed-off-by: swj <1186093704@qq.com>

### CODE REVIEW

File: pkg/object/s3.go [hunk 1]
```diff
@@ -396,6 +396,8 @@ func newS3(endpoint, accessKey, secretKey string) (ObjectStorage, error) {
 	}
 	if accessKey != "" {
 		awsConfig.Credentials = credentials.NewStaticCredentials(accessKey, secretKey, "")
+	} else if accessKey == "" && secretKey == "" {
+		awsConfig.Credentials = credentials.AnonymousCredentials
```
Comment (davies) [inline pkg/object/s3.go:400]: When the accessKey is empty, we usually use IAM role or other temporary token to access S3.

What's the current behavior to access public buckets? And, what's the use case that a public bucket is used with JuiceFS?

Comment (201341) [reply pkg/object/s3.go:400]: For public buckets. the juicefs sync can't downloads the file from s3. If the IAM is disabled and no temporary token , I can't use juicefs sync.

Comment (davies) [reply pkg/object/s3.go:400]: I see. The current patch will break IAM, we need another way to support anonymous access

Comment (201341) [reply pkg/object/s3.go:400]: ok, I will close the pr, and pull an issue.

**Pull request closed without merging**